src/ReTestItems.jl

```diff
@@ -849,6 +849,40 @@ end
 const GLOBAL_TEST_CONTEXT_FOR_TESTING = TestContext("ReTestItems", 0)
 const GLOBAL_TEST_SETUPS_FOR_TESTING = Dict{Symbol, TestSetup}()
 
+# Check the `skip` keyword, and return a `Bool` indicating if we should skip the testitem.
+# If `skip` is an expression, run it in a new module just like how we run testitems.
+# If the `skip` expression doesn't return a Bool, throw an informative error.
+function should_skip(ti::TestItem)
+    ti.skip isa Bool && return ti.skip
+    # `skip` is an expression.
+    # Give same scope as testitem body, e.g. imports should work.
+    skip_body = deepcopy(ti.skip::Expr)
+    softscope_all!(skip_body)
+    # Run in a new module to not pollute `Main`.
+    # Need to store the result of the `skip` expression so we can check it.
+    mod_name = gensym(Symbol(:skip_, ti.name))
+    skip_var = gensym(:skip)
+    skip_mod_expr = :(module $mod_name; $skip_var = $skip_body; end)
+    skip_mod = Core.eval(Main, skip_mod_expr)
+    # Check what the expression evaluated to.
+    skip = getfield(skip_mod, skip_var)
+    !isa(skip, Bool) && _throw_not_bool(ti, skip)
+    return skip::Bool
+end
+_throw_not_bool(ti, skip) = error("Test item $(repr(ti.name)) `skip` keyword must be a `Bool`, got `skip=$(repr(skip))`")
+
+# Log that we skipped the testitem, and record a "skipped" test result with empty stats.
+function skiptestitem(ti::TestItem, ctx::TestContext; verbose_results::Bool=true)
+    ts = DefaultTestSet(ti.name; verbose=verbose_results)
+    Test.record(ts, Test.Broken(:skipped, ti.name))
+    push!(ti.testsets, ts)
+    stats = PerfStats()
+    push!(ti.stats, stats)
+    log_testitem_skipped(ti, ctx.ntestitems)
+    return TestItemResult(ts, stats)
+end
```

Comment:
Should this be `!` since we're modifying `.testsets` and `.stats`? I always struggle with the border cases of this convention.

Reply:
i decided against it to match with `runtestitem` (which is unfortunately an exported name from a time before individual test-items auto-run in the repl)

i think it's just about acceptable because it modifies "private" fields (in contrast to "public" fields like `tags`, `name`, etc)